[PVR] Guide window: Go to date

ksooo opened the pull request:
This feature was several times requested in the forum. It adds a new menu item 'Go to date' to PVR Guide window's items:

![screenshot000](https://user-images.githubusercontent.com/3226626/34468205-23fc6b88-ef04-11e7-8dac-babf4148c169.png)
<img width="1680" alt="screen shot 2017-12-31 at 14 03 17" src="https://user-images.githubusercontent.com/3226626/34468204-23e42e6a-ef04-11e7-8edd-862a2745b35f.png">

@xhaggi mind taking a look?

ksooo committed:
[PVR] Guide window: Add 'Go to date' context menu item.

ksooo committed:
[PVR] Guide window: Add 'Go to' context menu items also for gap ('no information available') grid items.

ksooo committed:
[PVR] CGUIEPGGridContainerModel: Change method signatures to use const ref for params to avoid copy.

xhaggi commented on xbmc/pvr/windows/GUIWindowPVRGuide.cpp:
```diff
@@ -452,17 +454,46 @@ bool CGUIWindowPVRGuideBase::OnMessage(CGUIMessage& message)
   return bReturn || CGUIWindowPVRBase::OnMessage(message);
 }
 
+bool CGUIWindowPVRGuideBase::OnActionContextMenu()
```
We should rename the method or move the code to the case `ACTION_CONTEXT_MENU` because it's only used for gap items. At the moment, it's a little confusing because I thought it handles context menu actions for all items.

ksooo replied:
Do you have a suggestion for a better name?

ksooo commented:
@xhaggi I removed the method with the questionable name and moved the code to `CGUIWindowPVRGuideBase::OnMessage`. It's not called from different places, so no need for a separate method, imo. Okay now?

xhaggi reviewed (approved):
Better now, thanks.

da-anda commented:
since there are a lot of "go to..." items now - what would you guys think about grouping them into a secondary context menu level? Like on main level we only show "go to..." and then replace the context menu items with the 4 "go to" ones? So basically the same thing we do with the "Manage ..." context menu item on f.e. movie library items. Just a thought, no demand.

ksooo commented:
> since there are a lot of "go to..." items now - what would you guys think about grouping them into a secondary context menu level?

 Hehe, I actually had that implemented and played around with it. But due to the additional click necessary to access the actual "go to" entry of interest it somehow felt wrong to me, because the "go to" actions are shortcuts for people who want to save time. Otherwise they would just scroll to the epg grid. Adding another step just for cosmetically purposes is imo not the right approach here.

da-anda commented:
the one more click argument doesn't really hold up, because when grouped you can reach the "go to beginning" item actually way faster by clicking "up + select + select" - so just 3 clicks instead of 5. Also, it'll always be faster than manually navigating the EGP, regardless if it's one more click or not. But whatever.

ksooo commented:
That implies rust to go to menu is on top ... and this is technically currently not possible - at least to my knowledge.

da-anda commented:
you can press up in a context menu to get to the last item. And when the grouped "go to..." item is the last item, all it takes is `up + select + select` to get to the top most "go to" function. Just saying
